### PR TITLE
fix(ship): fixed for when user is (magically) undefined

### PR DESCRIPTION
**Pull request by favna**
Some odd edge case that occurrs in Ship sometimes apparently.

### CODE REVIEW

**Review by kyranet — DISMISSED** [review]
This is a wrong fix. It should be handled wherever this is called and never be called with null or undefined.

File: src/commands/Misc/ship.ts [hunk 1]
```diff
@@ -53,8 +53,11 @@ export default class extends SkyraCommand {
 
 	public async run(message: KlasaMessage, [firstUser, secondUser]: [KeyedMemberTag, KeyedMemberTag]) {
 		// We need the UserTags to resolve the avatar and username
-		const firstUserTag = this.client.userTags.get(firstUser.id)!;
-		const secondUserTag = this.client.userTags.get(secondUser.id)!;
+		let firstUserTag = this.client.userTags.get(firstUser.id);
+		let secondUserTag = this.client.userTags.get(secondUser.id);
+
+		if (!firstUserTag) firstUserTag = { avatar: null, discriminator: '0001', bot: false, username: '' };
+		if (!secondUserTag) secondUserTag = { avatar: null, discriminator: '0001', bot: false, username: '' };
```
Comment (kyranet) [inline src/commands/Misc/ship.ts]:
```suggestion
		const firstUserTag = this.client.userTags.get(firstUser.id) ?? { avatar: null, discriminator: '0001', bot: false, username: '' };
		const secondUserTag = this.client.userTags.get(secondUser.id) ?? { avatar: null, discriminator: '0001', bot: false, username: '' };
```

Comment (kyranet) [reply src/commands/Misc/ship.ts]: Or even better:

```ts
private readonly kEmptyUserTag = { avatar: null, discriminator: '0001', bot: false, username: '' };
```

And then:

```suggestion
		const firstUserTag = this.client.userTags.get(firstUser.id) ?? this.kEmptyUserTag;
		const secondUserTag = this.client.userTags.get(secondUser.id) ?? this.kEmptyUserTag;
```

**Review by kyranet — DISMISSED** [review]

**Comment by favna** [comment]
So about the ship problem @kyranet, the empty user tag thing wouldn't really work because the usertag is also being read for the  text and such. I have 2 ideas rn
1. (currently coded): pick a random usertag based on a randomKey of the memberTags, to ensure that the random usertag is in the current guild at least.
2. default to skyra lol

**Comment by kyranet** [comment]
3. use fetch instead of get

**Review by kyranet — APPROVED** [review]

**Pull request merged**